examples/plot_ECoG_preprocessing_tutorial.py

```diff
@@ -0,0 +1,242 @@
+plt.xlabel('Samples (0.1 sec of data)')
+plt.ylabel('Amplitude (au)')
+plt.title('One channel of neural data')
+pass
+
+
+# %% 
+# Resampling in the frequency domain
+# ------------------------
+# - Resample the data down to make computations much faster downstream
+# - The resampling algorithm is done in the frequency domain
+
+from process_nwb import resample
+
+# %%
+#
+
+new_sample_rate = 1000 #hz
+rs_data = resample.resample(neural_data, new_sample_rate, sample_rate, real=True, axis=0)
+
+plt.plot(rs_data[:100,0])
+plt.xlabel('Samples (0.1 sec of data)')
+plt.ylabel('Amplitude (au)')
+plt.title('One channel of neural data')
+pass
+
+# %% 
+# Re-referencing with common average referencing
+# ------------------------
+# - Often times there is common noise from neural recording such as movement artifact or line noise
+# - To remove, a robust estimate of the average across all signals (for each timepoint) is calculated than subtracted all channels
+
+car_data = ecog.common_referencing.subtract_CAR(rs_data, mean_frac=.95, round_func=np.ceil)
+
+plt.plot(car_data[:100,0])
+plt.xlabel('Samples (0.1 sec of data)')
+plt.ylabel('Amplitude (au)')
+plt.title('One channel of neural data')
+pass
+
+# %%
+#  Notch filtering
+# ------------------------
+# - Since line noise is not of equal strength across channels, CAR fails at getting rid of all line noise. 
+# - Therefore, notch filtering is used to remove line noise on all channels
+
+nth_data = ecog.linenoise_notch.apply_linenoise_notch(car_data, new_sample_rate, fft=True)
+
+plt.plot(nth_data[:100,0])
+plt.xlabel('Samples (0.1 sec of data)')
+plt.ylabel('Amplitude (au)')
+plt.title('One channel of neural data')
+pass
+
+# %%
+#
+
+from scipy.signal import welch
+from scipy.signal.windows import kaiser
+
+# %% 
+#
+
+
+window = kaiser(1024, 2)
+freq, car_pwr = welch(car_data[:,0], fs=new_sample_rate, window=window)
+_, nth_pwr = welch(nth_data[:,0], fs=new_sample_rate, window=window)
+
+fig, axs = plt.subplots(1,2, figsize=(10,4), sharey=True, sharex=True)
+#fig.tight_layout()
+axs[0].semilogy(freq, car_pwr)
+axs[0].set_xlabel('Frequency (Hz)')
+axs[0].set_ylabel('Power density (V^2/Hz)')
+axs[0].set_xlim([1, 150])
+axs[0].set_title('Pre notch filtering')
+
+axs[1].semilogy(freq, nth_pwr)
+axs[1].set_xlabel('Frequency (Hz)')
+axs[1].set_xlim([1, 150])
+axs[1].set_title('Post notch filtering')
+fig.tight_layout()
+pass
+
+
+# %%
+#  Time-frequency decomposition with wavelets
+# ------------------------
+# - Here we decompose the neural time series into 6 different frequency bands using a wavelet transform
+# - The wavelet transform outputs the analytical amplitude, which is complex valued.
+# - I'm plotting the magnitude of analytical amplitude which is a proxy of band power
+
+from process_nwb import wavelet_transform
+
+# %% 
+#
+
+
+tf_data, _, ctr_freq, bw = wavelet_transform.wavelet_transform(nth_data, new_sample_rate, filters='rat', hg_only=True, X_fft_h=None, npad=None)
+
```

Comment:
```suggestion
tf_data = abs(tf_data)
```

I would just take the ABS here. You should be zscoring the amplitude, not the complex waveform.

Reply:
This works as intended for the first graph

Reply:
After applying abs first before further processing